docs/backport-workflow.md

```diff
@@ -0,0 +1,526 @@
+# Automated Backport Workflow
+
+This document describes the automated backport workflow for the LND project.
+
+## Table of Contents
+
+1. [Overview](#overview)
+2. [How to Use](#how-to-use)
+3. [Workflow Triggers](#workflow-triggers)
+4. [Label Format](#label-format)
+5. [Workflow Steps](#workflow-steps)
+6. [Handling Conflicts](#handling-conflicts)
+7. [Multiple Backports](#multiple-backports)
+8. [Technical Details](#technical-details)
+9. [Troubleshooting](#troubleshooting)
+
+## Overview
+
+The automated backport workflow simplifies the process of backporting merged PRs from the `master` branch to release branches (e.g., `v0.20.x-branch`, `v0.19.x-branch`).
+
+Instead of manually creating branches, cherry-picking commits, and creating PRs, maintainers can simply add a label to the master PR, and the workflow handles the rest.
+
+## How to Use
+
+### Basic Usage
+
+1. **Merge a PR to master** (or have it already merged)
+2. **Add a backport label** in the format: `backport-v<version>-branch`
+   - Example: `backport-v0.20.x-branch`
+3. **The workflow automatically**:
+   - Validates the target branch exists
+   - Cherry-picks the commits
+   - Creates a new PR targeting the release branch
+   - Adds the `no-changelog` label (since release notes are in the master PR)
+
+### Example Scenario
+
+```
+Day 1, 10:00 - PR #1234 "Fix critical bug" merged to master
+Day 1, 10:30 - Add label: backport-v0.20.x-branch
+Day 1, 10:31 - Workflow creates PR #1235 automatically
+              Title: "[v0.20.x-branch] Backport #1234: Fix critical bug"
+              Base: v0.20.x-branch
+              Labels: no-changelog
+Day 1, 14:00 - Maintainer reviews and merges PR #1235
+```
+
+## Workflow Triggers
+
+The backport workflow triggers in two scenarios:
+
+### Scenario 1: Label Before Merge
+
+```
+1. Open PR #1234
+2. Add label: backport-v0.20.x-branch
+3. Review and approve PR
+4. Merge PR #1234
+5. → Workflow triggers on PR close event
+6. → Backport PR #1235 created immediately
+```
+
+### Scenario 2: Label After Merge (Recommended)
```

Comment:
What's the reason for this to be a recommended approach?

Reply:
removed the recommendation part.